assets/src/edit-story/utils/test/cleanForSlug.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import cleanForSlug from '../cleanForSlug';
+
+describe( 'cleanForSlug', () => {
+	it( 'should return string prepared for use as URL slug', () => {
+		expect( cleanForSlug( ' /Déjà_vu. ' ) ).toBe( 'deja-vu' );
+	} );
+
+	it( 'should return an empty string for missing argument', () => {
+		expect( cleanForSlug() ).toBe( '' );
+	} );
+
+	it( 'should return an empty string for falsy argument', () => {
+		expect( cleanForSlug( null ) ).toBe( '' );
+	} );
+
+	it( 'should trim leading whitespace', () => {
+		const result = cleanForSlug( ' Hello World ' );
+		expect( result ).toStrictEqual( 'hello-world' );
+	} );
+
+	it( 'should trim leading and ending dashes', () => {
+		const result = cleanForSlug( '-Hello-World-' );
+		expect( result ).toStrictEqual( 'hello-world' );
+	} );
+
+	it( 'should remove diacritics', () => {
+		const result = cleanForSlug( '-Hello-World-' );
+		expect( result ).toStrictEqual( 'hello-world' );
+	} );
+
+	// Disable reason: not implemented yet.
+	// eslint-disable-next-line jest/no-disabled-tests
+	it.skip( 'should replace umlauts if locale is de-*', () => {
+		expect( cleanForSlug( 'Übergrössengeschäft' ) ).toBe( 'uebergroessengeschaeft' );
```

Comment:
I suppose that should be for any language that uses umlauts (like Estonian ;) ) and not just for de-*?

Reply:
Yeah, it would be implemented that way. Still todo.